extensions/scarb-doc/src/types.rs

```diff
@@ -286,15 +408,118 @@ pub struct Trait {
     pub id: TraitId,
     pub node: ast::ItemTraitPtr,
 
+    pub trait_constants: Vec<TraitConstant>,
+    pub trait_types: Vec<TraitType>,
+    pub trait_functions: Vec<TraitFunction>,
+
     pub item_data: ItemData,
+    pub lookup_item_data: LookupItemData,
 }
 
 impl Trait {
-    pub fn new(db: &dyn DefsGroup, id: TraitId, node: &ast::ItemTrait) -> Self {
+    pub fn new(db: &RootDatabase, id: TraitId) -> Self {
+        let trait_constants = db.trait_constants(id).unwrap();
+        let trait_constants = trait_constants
+            .iter()
+            .map(|(_name, trait_constant_id)| TraitConstant::new(db, *trait_constant_id))
+            .collect::<Vec<_>>();
+
+        let trait_types = db.trait_types(id).unwrap();
+        let trait_types = trait_types
+            .iter()
+            .map(|(_name, trait_type_id)| TraitType::new(db, *trait_type_id))
+            .collect::<Vec<_>>();
+
+        let trait_functions = db.trait_functions(id).unwrap();
+        let trait_functions = trait_functions
+            .iter()
+            .map(|(_name, trait_function_id)| TraitFunction::new(db, *trait_function_id))
+            .collect::<Vec<_>>();
+
+        let node = id.stable_ptr(db);
+        Self {
+            id,
+            node,
+            trait_constants,
+            trait_types,
+            trait_functions,
+            item_data: ItemData::new(db, id, node),
+            lookup_item_data: LookupItemData::new(
+                db,
+                LookupItemId::ModuleItem(ModuleItemId::Trait(id)),
+            ),
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct TraitConstant {
+    pub id: TraitConstantId,
+    pub node: ast::TraitItemConstantPtr,
+
+    pub item_data: ItemData,
+    pub lookup_item_data: LookupItemData,
+}
```

Comment:
Why this can't be a regular constant? 🤔

Reply:
I guess this applies to all of the nested things inside of the traits

Reply:
Cairo stuff I guess